storey/sources.py

```diff
@@ -63,14 +63,14 @@ def _build_event(self, element, key, event_time):
             if key:
                 event.key = key
             elif self._key_field:
-                event.key = event.body[self._key_field]
+                event.key = get_key_from_data(event.body, self._key_field)
```

Comment:
i can't add this comment in the right place
need to change the signature of __init__ func that key_field is a union